### PR TITLE
refactor(ui): add social landing page

### PR DESCRIPTION
**Pull request by simeng-li**
## Summary

Add social landing page.

### Context:
Alipayay has a strict size limit on state params
Previously we store all platform and native callback link data in a state as a JSON object. Which violates the Alipay's state length limitation. 
We have to find a way to pass the native callback link from webview to the callback page. 

### Approach
Build a connector landing page to store the native callbackLink in session storage. So the callback page can get the link later. 

1. Extract a SocialLanding Container as the UI part is shared by Callback and the new SocialLanding page. 
2. Build the new SocialLanding page, with path `/social-landing`
3. Simplify the usage of state, only return a UUID as it supposes to be. 
4. In useSocial handler, redirect to the landing page first if is a native universal sign in flow, else redirect to the given URL directly. 
5. Add a new use-social-landing hook to handle the logic for the landing page. 
- read callbackLink from SearchParams and store in session storage
- redirect to the original redirectUri
6. In the callback handler, load the callback link from session storage first, if exist goes to the callback link, if not goes to the /sign-in/callback link directly.

image

## Linear Issue Reference

log-2727

## Testing

test locally
@logto-io/eng

### CODE REVIEW

**Review by gao-sun — APPROVED** [review]
good

File: packages/ui/src/App.tsx [hunk 1]
```diff
@@ -55,12 +56,16 @@ const App = () => {
...
-            <Route path="/sign-in/callback/:connector" element={<SocialSignInCallback />} />
             <Route path="/sign-in/:method" element={<SecondarySignIn />} />
             <Route path="/register" element={<Register />} />
             <Route path="/register/:method" element={<Register />} />
+
+            {/* social sign-in pages */}
+            <Route path="/sign-in/callback/:connector" element={<SocialSignInCallback />} />
             <Route path="/callback/:connector" element={<Callback />} />
             <Route path="/social-register/:connector" element={<SocialRegister />} />
+            <Route path="/social-landing/:connector" element={<SocialLanding />} />
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/App.tsx:67]: kool. will update later

File: packages/ui/src/containers/SocialLanding/index.module.scss [hunk 2]
```diff
@@ -0,0 +1,22 @@
+@use '@/scss/underscore' as _;
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/containers/SocialLanding/index.module.scss:1]: image

 current design we have

File: packages/phrases/src/locales/en.ts [hunk 3]
```diff
@@ -81,12 +81,12 @@ const translation = {
...
       invalid_email: 'The email is invalid',
       invalid_phone: 'The phone number is invalid',
-      invalid_connector_auth: 'The authorization is invalid',
-      missing_auth_data: 'The authorization code or state is missing',
       password_min_length: 'Password requires a minimum of {{min}} characters.',
       passwords_do_not_match: 'Passwords do not match.',
       agree_terms_required: 'You must agree to the Terms of Use before continuing.',
       invalid_passcode: 'The passcode is invalid.',
+      invalid_connector_auth: 'The authorization is invalid',
+      invalid_connector_request: 'The connector data is invalid',
```
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/phrases/src/locales/en.ts]: updated

File: packages/ui/src/containers/SocialLanding/index.module.scss (hunk 2 above)
[the remark this answers is not in the document]
Comment (simeng-li) [reply packages/ui/src/containers/SocialLanding/index.module.scss:1]: will double check with GM

**Pull request merged**